angular/projects/admin-nrpti/src/app/mines/mines-rows/mines-table-row.component.ts

```diff
@@ -0,0 +1,53 @@
+import { Component, HostListener, OnInit, ChangeDetectorRef } from '@angular/core';
+import { Router } from '@angular/router';
+import { TableRowComponent } from 'nrpti-angular-components';
+import { Entity } from '../../../../../common/src/app/models/master/common-models/entity';
+
+@Component({
+  selector: 'tr[app-mines-table-row]',
+  templateUrl: './mines-table-row.component.html',
+  styleUrls: ['./mines-table-row.component.scss']
+})
+export class MinesTableRowComponent extends TableRowComponent implements OnInit {
+  public dropdownItems = ['Edit', 'Delete'];
+
+  public entityString = '';
+
+  constructor(private router: Router, public changeDetectionRef: ChangeDetectorRef) {
+    super();
+  }
+
+  ngOnInit() {
+    this.populateTextFields();
+
+    this.changeDetectionRef.detectChanges();
+  }
+
+  populateTextFields() {
+    if (this.rowData && this.rowData.issuedTo) {
+      this.entityString = new Entity(this.rowData.issuedTo).getEntityNameString();
+    }
+  }
+
+  @HostListener('click') onItemClicked() {
+    switch (this.rowData._schemaName) {
+      case 'Mine':
+        this.router.navigate(['mines', 'mines', this.rowData._id, 'detail']);
+        break;
+      default:
+        break;
+    }
+  }
+
+  edit(event: Event) {
+    event.stopPropagation();
+
+    switch (this.rowData._schemaName) {
+      case 'Mine':
+        this.router.navigate(['mines', 'mines', this.rowData._id, 'edit']);
+        break;
+      default:
+        break;
+    }
+  }
+}
```

Comment:
Not sure why we're switching here when an if statement will do.

Reply:
This is the pattern we used in other places, which I copied over.
Can change it if its important

Reply:
I don't think switches with a single item make sense.  It's not really achieving anything other than adding to vertical space wasted.

Reply:
sure, will put up a fix in 2 mins

Reply:
updated